src/structures/CommandInteractionOptionResolver.js

```diff
@@ -0,0 +1,169 @@
+   * @returns {?GuildChannel|APIInteractionDataResolvedChannel}
+   * The value of the option, or null if not set and not required.
+   */
+  getChannel(name, required = false) {
+    const option = this._getTypedOption(name, ['CHANNEL'], ['channel'], required);
+    return option?.channel ?? null;
+  }
+
+  /**
+   * Gets a string option.
+   * @param {string} name The name of the option.
+   * @param {boolean} required Whether to throw an error if the option is not found.
+   * @returns {?string} The value of the option, or null if not set and not required.
+   */
+  getString(name, required = false) {
+    const option = this._getTypedOption(name, ['STRING'], ['value'], required);
+    return option?.value ?? null;
+  }
+
+  /**
+   * Gets an integer option.
+   * @param {string} name The name of the option.
+   * @param {boolean} required Whether to throw an error if the option is not found.
+   * @returns {?number} The value of the option, or null if not set and not required.
+   */
+  getInteger(name, required = false) {
+    const option = this._getTypedOption(name, ['INTEGER'], ['value'], required);
+    return option?.value ?? null;
+  }
+
+  /**
+   * Gets a user option.
+   * @param {string} name The name of the option.
+   * @param {boolean} required Whether to throw an error if the option is not found.
+   * @returns {?User} The value of the option, or null if not set and not required.
+   */
+  getUser(name, required = false) {
+    const option = this._getTypedOption(name, ['USER'], ['user'], required);
+    return option?.user ?? null;
+  }
+
+  /**
+   * Gets a member option.
+   * @param {string} name The name of the option.
+   * @param {boolean} required Whether to throw an error if the option is not found.
+   * @returns {?GuildMember|APIInteractionDataResolvedGuildMember}
+   * The value of the option, or null if not set and not required.
+   */
+  getMember(name, required = false) {
+    const option = this._getTypedOption(name, ['MEMBER'], ['member'], required);
+    return option?.member ?? null;
+  }
+
+  /**
+   * Gets a role option.
+   * @param {string} name The name of the option.
+   * @param {boolean} required Whether to throw an error if the option is not found.
+   * @returns {?Role|APIRole} The value of the option, or null if not set and not required.
+   */
+  getRole(name, required = false) {
+    const option = this._getTypedOption(name, ['ROLE'], ['role'], required);
+    return option?.role ?? null;
+  }
+
+  /**
+   * Gets a mentionable option.
+   * @param {string} name The name of the option.
+   * @param {boolean} required Whether to throw an error if the option is not found.
+   * @returns {?User|GuildMember|Role} The value of the option, or null if not set and not required.
+   */
+  getMentionable(name, required = false) {
+    const option = this._getTypedOption(name, ['MENTIONABLE'], ['user', 'member', 'role'], required);
+    return option?.member ?? option?.user ?? option?.role ?? null;
```

Comment:
```suggestion
    return option && (option.member ?? option.user ?? option.role ?? null);
```

Reply:
This makes the code harder to read imo

Reply:
eh, i feel like it makes more sense here since if `option` is null, you're just doing `undefined ?? undefined ?? undefined ?? null`